external builds fail due to undef ref to embedded model symbols

running
```bash
git clone https://github.com/Netflix/vmaf
meson setup --buildtype release -Dlibdir=lib vmaf-build vmaf/libvmaf
sudo -E meson install -C vmaf-build
```
fails on meson install with the following:
```c
[93/120] Linking target src/libvmaf.so.1.0.0.
FAILED: src/libvmaf.so.1.0.0 
c++  -o src/libvmaf.so.1.0.0 'src/25a6634@@vmaf@sha/meson-generated_.._vmaf_v0.6.1.json.c.o' 'src/25a6634@@vmaf@sha/meson-generated_.._vmaf_b_v0.6.3.json.c.o' 'src/25a6634@@vmaf@sha/meson-generated_.._vmaf_v0.6.1neg.json.c.o' 'src/25a6634@@vmaf@sha/libvmaf.c.o' 'src/25a6634@@vmaf@sha/compute_vmaf.c.o' 'src/25a6634@@vmaf@sha/predict.c.o' 'src/25a6634@@vmaf@sha/model.c.o' 'src/25a6634@@vmaf@sha/svm.cpp.o' 'src/25a6634@@vmaf@sha/picture.c.o' 'src/25a6634@@vmaf@sha/mem.c.o' 'src/25a6634@@vmaf@sha/output.c.o' 'src/25a6634@@vmaf@sha/fex_ctx_vector.c.o' 'src/25a6634@@vmaf@sha/thread_pool.c.o' 'src/25a6634@@vmaf@sha/dict.c.o' 'src/25a6634@@vmaf@sha/opt.c.o' 'src/25a6634@@vmaf@sha/ref.c.o' 'src/25a6634@@vmaf@sha/read_json_model.c.o' 'src/25a6634@@vmaf@sha/pdjson.c.o' 'src/25a6634@@x86_avx2@sta/feature_common_convolution_avx.c.o' 'src/25a6634@@x86_avx2@sta/feature_x86_motion_avx2.c.o' 'src/25a6634@@x86_avx2@sta/feature_x86_vif_avx2.c.o' 'src/25a6634@@x86_avx2@sta/feature_x86_adm_avx2.c.o' 'src/25a6634@@libvmaf_feature@sta/feature_picture_copy.c.o' 'src/25a6634@@libvmaf_feature@sta/feature_integer_psnr.c.o' 'src/25a6634@@libvmaf_feature@sta/feature_third_party_xiph_psnr_hvs.c.o' 'src/25a6634@@libvmaf_feature@sta/feature_feature_extractor.c.o' 'src/25a6634@@libvmaf_feature@sta/feature_alias.c.o' 'src/25a6634@@libvmaf_feature@sta/feature_integer_adm.c.o' 'src/25a6634@@libvmaf_feature@sta/feature_feature_collector.c.o' 'src/25a6634@@libvmaf_feature@sta/feature_integer_motion.c.o' 'src/25a6634@@libvmaf_feature@sta/feature_integer_vif.c.o' 'src/25a6634@@libvmaf_feature@sta/feature_ciede.c.o' 'src/25a6634@@libvmaf_feature@sta/feature_common_alignment.c.o' 'src/25a6634@@libvmaf_feature@sta/feature_float_ssim.c.o' 'src/25a6634@@libvmaf_feature@sta/feature_float_ms_ssim.c.o' 'src/25a6634@@libvmaf_feature@sta/feature_ssim.c.o' 'src/25a6634@@libvmaf_feature@sta/feature_ms_ssim.c.o' 'src/25a6634@@libvmaf_feature@sta/feature_iqa_decimate.c.o' 'src/25a6634@@libvmaf_feature@sta/feature_iqa_ssim_tools.c.o' 'src/25a6634@@libvmaf_feature@sta/feature_iqa_math_utils.c.o' 'src/25a6634@@libvmaf_feature@sta/feature_iqa_convolve.c.o' 'src/25a6634@@libvmaf_cpu@sta/cpuid.obj' 'src/25a6634@@libvmaf_cpu@sta/cpu.c.o' 'src/25a6634@@libvmaf_cpu@sta/x86_cpu.c.o' -Wl,--as-needed -Wl,--no-undefined -Wl,-O1 -shared -fPIC -Wl,--start-group -Wl,-soname,libvmaf.so.1 -pthread -lm -Wl,--end-group
/usr/bin/ld: src/25a6634@@vmaf@sha/model.c.o:(.data.rel.ro+0x8): undefined reference to `___src_______model_vmaf_v0_6_1_json'
/usr/bin/ld: src/25a6634@@vmaf@sha/model.c.o:(.data.rel.ro+0x10): undefined reference to `___src_______model_vmaf_v0_6_1_json_len'
/usr/bin/ld: src/25a6634@@vmaf@sha/model.c.o:(.data.rel.ro+0x20): undefined reference to `___src_______model_vmaf_b_v0_6_3_json'
/usr/bin/ld: src/25a6634@@vmaf@sha/model.c.o:(.data.rel.ro+0x28): undefined reference to `___src_______model_vmaf_b_v0_6_3_json_len'
/usr/bin/ld: src/25a6634@@vmaf@sha/model.c.o:(.data.rel.ro+0x38): undefined reference to `___src_______model_vmaf_v0_6_1neg_json'
/usr/bin/ld: src/25a6634@@vmaf@sha/model.c.o:(.data.rel.ro+0x40): undefined reference to `___src_______model_vmaf_v0_6_1neg_json_len'
collect2: error: ld returned 1 exit status
...

[95/120] Linking target tools/vmafossexec.
FAILED: tools/vmafossexec 
c++  -o tools/vmafossexec 'tools/f9d35d4@@vmafossexec@exe/main.cpp.o' 'tools/f9d35d4@@vmafossexec@exe/read_frame.c.o' -Wl,--as-needed -Wl,--no-undefined -Wl,-O1 -Wl,--start-group src/libvmaf.a -pthread -lm -Wl,--end-group '-Wl,-rpath,$ORIGIN/../src' -Wl,-rpath-link,/code/vmaf-build/src
/usr/bin/ld: src/libvmaf.a(model.c.o):(.data.rel.ro+0x8): undefined reference to `___src_______model_vmaf_v0_6_1_json'
/usr/bin/ld: src/libvmaf.a(model.c.o):(.data.rel.ro+0x10): undefined reference to `___src_______model_vmaf_v0_6_1_json_len'
/usr/bin/ld: src/libvmaf.a(model.c.o):(.data.rel.ro+0x20): undefined reference to `___src_______model_vmaf_b_v0_6_3_json'
/usr/bin/ld: src/libvmaf.a(model.c.o):(.data.rel.ro+0x28): undefined reference to `___src_______model_vmaf_b_v0_6_3_json_len'
/usr/bin/ld: src/libvmaf.a(model.c.o):(.data.rel.ro+0x38): undefined reference to `___src_______model_vmaf_v0_6_1neg_json'
/usr/bin/ld: src/libvmaf.a(model.c.o):(.data.rel.ro+0x40): undefined reference to `___src_______model_vmaf_v0_6_1neg_json_len'
collect2: error: ld returned 1 exit status
[96/120] Linking target tools/vmaf.
FAILED: tools/vmaf 
c++  -o tools/vmaf 'tools/f9d35d4@@vmaf@exe/vmaf.c.o' 'tools/f9d35d4@@vmaf@exe/cli_parse.c.o' 'tools/f9d35d4@@vmaf@exe/y4m_input.c.o' 'tools/f9d35d4@@vmaf@exe/vidinput.c.o' 'tools/f9d35d4@@vmaf@exe/yuv_input.c.o' -Wl,--as-needed -Wl,--no-undefined -Wl,-O1 -Wl,--start-group src/libvmaf.a -pthread -lm -Wl,--end-group '-Wl,-rpath,$ORIGIN/../src' -Wl,-rpath-link,/code/vmaf-build/src
/usr/bin/ld: src/libvmaf.a(model.c.o):(.data.rel.ro+0x8): undefined reference to `___src_______model_vmaf_v0_6_1_json'
/usr/bin/ld: src/libvmaf.a(model.c.o):(.data.rel.ro+0x10): undefined reference to `___src_______model_vmaf_v0_6_1_json_len'
/usr/bin/ld: src/libvmaf.a(model.c.o):(.data.rel.ro+0x20): undefined reference to `___src_______model_vmaf_b_v0_6_3_json'
/usr/bin/ld: src/libvmaf.a(model.c.o):(.data.rel.ro+0x28): undefined reference to `___src_______model_vmaf_b_v0_6_3_json_len'
/usr/bin/ld: src/libvmaf.a(model.c.o):(.data.rel.ro+0x38): undefined reference to `___src_______model_vmaf_v0_6_1neg_json'
/usr/bin/ld: src/libvmaf.a(model.c.o):(.data.rel.ro+0x40): undefined reference to `___src_______model_vmaf_v0_6_1neg_json_len'
collect2: error: ld returned 1 exit status
[98/120] Compiling C++ object 'test/9f86d08@@test_predict@exe/.._src_svm.cpp.o'.
ninja: build stopped: subcommand failed.
Could not rebuild vmaf-build/
```

grepping
```c
grep -R ___src_______model_vmaf_v0_6_1_json vmaf-build/ 
grep: vmaf-build/src/libvmaf.so: No such file or directory
Binary file vmaf-build/src/25a6634@@vmaf@sha/model.c.o matches
Binary file vmaf-build/src/libvmaf.a matches
grep: vmaf-build/src/libvmaf.so.1: No such file or directory
Binary file vmaf-build/test/9f86d08@@test_predict@exe/.._src_model.c.o matches
Binary file vmaf-build/test/9f86d08@@test_model@exe/test_model.c.o matches
```

running `ninja -v`, I see
```c
ninja: Entering directory `vmaf-build/'
[1/120] /usr/bin/xxd --include ../vmaf/libvmaf/src/../../model/vmaf_v0.6.1.json src/vmaf_v0.6.1.json.c
[2/120] /usr/bin/xxd --include ../vmaf/libvmaf/src/../../model/vmaf_v0.6.1neg.json src/vmaf_v0.6.1neg.json.c
...
```

--- a/libvmaf/meson.build
+++ b/libvmaf/meson.build
@@ -4,6 +4,7 @@ project('libvmaf', ['c', 'cpp'],
                        'cpp_std=c++11',
                        'warning_level=2',
                        'buildtype=release',
+                       'default_library=both',
                       ],
     meson_version: '>= 0.47.0')
 

--- a/libvmaf/src/meson.build
+++ b/libvmaf/src/meson.build
@@ -122,7 +122,11 @@ if built_in_models_enabled
             json_model_c_sources += custom_target(
                   model_file,
                   output : '@PLAINNAME@.c',
-                  input : model_dir + model_file,
+                  input : configure_file(
+                    input : model_dir + model_file,
+                    output : model_file,
+                    copy: true
+                  ),
                   command : [xxd, '--include', '@INPUT@', '@OUTPUT@'],
             )
         endforeach

--- a/libvmaf/src/meson.build
+++ b/libvmaf/src/meson.build
@@ -297,7 +297,7 @@ else
     vmaf_soversion = vmaf_api_version_major
 endif
 
-libvmaf = both_libraries(
+libvmaf = library(
     'vmaf',
     [libvmaf_sources, rev_target, json_model_c_sources],
     include_directories : [libvmaf_inc, vmaf_include],

--- a/libvmaf/src/model.c
+++ b/libvmaf/src/model.c
@@ -21,19 +21,19 @@ typedef struct VmafBuiltInModel {
 
 #if VMAF_BUILT_IN_MODELS
 #if VMAF_FLOAT_FEATURES
-extern const char ___src_______model_vmaf_float_v0_6_1neg_json;
-extern const int ___src_______model_vmaf_float_v0_6_1neg_json_len;
-extern const char ___src_______model_vmaf_float_v0_6_1_json;
-extern const int ___src_______model_vmaf_float_v0_6_1_json_len;
-extern const char ___src_______model_vmaf_float_b_v0_6_3_json;
-extern const int ___src_______model_vmaf_float_b_v0_6_3_json_len;
+extern const char src_vmaf_float_v0_6_1neg_json;
+extern const int src_vmaf_float_v0_6_1neg_json_len;
+extern const char src_vmaf_float_v0_6_1_json;
+extern const int src_vmaf_float_v0_6_1_json_len;
+extern const char src_vmaf_float_b_v0_6_3_json;
+extern const int src_vmaf_float_b_v0_6_3_json_len;
 #endif
-extern const char ___src_______model_vmaf_v0_6_1_json;
-extern const int ___src_______model_vmaf_v0_6_1_json_len;
-extern const char ___src_______model_vmaf_b_v0_6_3_json;
-extern const int ___src_______model_vmaf_b_v0_6_3_json_len;
-extern const char ___src_______model_vmaf_v0_6_1neg_json;
-extern const int ___src_______model_vmaf_v0_6_1neg_json_len;
+extern const char src_vmaf_v0_6_1_json;
+extern const int src_vmaf_v0_6_1_json_len;
+extern const char src_vmaf_b_v0_6_3_json;
+extern const int src_vmaf_b_v0_6_3_json_len;
+extern const char src_vmaf_v0_6_1neg_json;
+extern const int src_vmaf_v0_6_1neg_json_len;
 #endif
 
 static const VmafBuiltInModel built_in_models[] = {
@@ -41,34 +41,34 @@ static const VmafBuiltInModel built_in_models[] = {
 #if VMAF_FLOAT_FEATURES
     {
         .version = "vmaf_float_v0.6.1neg",
-        .data = &___src_______model_vmaf_float_v0_6_1neg_json,
-        .data_len = &___src_______model_vmaf_float_v0_6_1neg_json_len,
+        .data = &src_vmaf_float_v0_6_1neg_json,
+        .data_len = &src_vmaf_float_v0_6_1neg_json_len,
     },
     {
         .version = "vmaf_float_v0.6.1",
-        .data = &___src_______model_vmaf_float_v0_6_1_json,
-        .data_len = &___src_______model_vmaf_float_v0_6_1_json_len,
+        .data = &src_vmaf_float_v0_6_1_json,
+        .data_len = &src_vmaf_float_v0_6_1_json_len,
     },
     {
         .version = "vmaf_float_b_v0.6.3",
-        .data = &___src_______model_vmaf_float_b_v0_6_3_json,
-        .data_len = &___src_______model_vmaf_float_b_v0_6_3_json_len,
+        .data = &src_vmaf_float_b_v0_6_3_json,
+        .data_len = &src_vmaf_float_b_v0_6_3_json_len,
     },
 #endif
     {
         .version = "vmaf_v0.6.1",
-        .data = &___src_______model_vmaf_v0_6_1_json,
-        .data_len = &___src_______model_vmaf_v0_6_1_json_len,
+        .data = &src_vmaf_v0_6_1_json,
+        .data_len = &src_vmaf_v0_6_1_json_len,
     },
     {
         .version = "vmaf_b_v0.6.3",
-        .data = &___src_______model_vmaf_b_v0_6_3_json,
-        .data_len = &___src_______model_vmaf_b_v0_6_3_json_len,
+        .data = &src_vmaf_b_v0_6_3_json,
+        .data_len = &src_vmaf_b_v0_6_3_json_len,
     },
     {
         .version = "vmaf_v0.6.1neg",
-        .data = &___src_______model_vmaf_v0_6_1neg_json,
-        .data_len = &___src_______model_vmaf_v0_6_1neg_json_len,
+        .data = &src_vmaf_v0_6_1neg_json,
+        .data_len = &src_vmaf_v0_6_1neg_json_len,
     },
 #endif
     { 0 }

--- a/libvmaf/test/meson.build
+++ b/libvmaf/test/meson.build
@@ -7,7 +7,7 @@ test_inc = include_directories('.')
 test_context = executable('test_context',
     ['test.c', 'test_context.c'],
     include_directories : [libvmaf_inc, test_inc],
-    link_with : libvmaf.get_static_lib(),
+    link_with : get_option('default_library') == 'both' ? libvmaf.get_static_lib() : libvmaf,
     dependencies:[stdatomic_dependency],
 )
 

--- a/libvmaf/tools/meson.build
+++ b/libvmaf/tools/meson.build
@@ -10,7 +10,7 @@ vmafossexec = executable(
     dependencies: [stdatomic_dependency],
     c_args : vmaf_cflags_common,
     cpp_args : vmaf_cflags_common,
-    link_with : libvmaf.get_static_lib(),
+    link_with : get_option('default_library') == 'both' ? libvmaf.get_static_lib() : libvmaf,
     install : false,
 )
 
@@ -20,6 +20,6 @@ vmaf = executable(
     include_directories : [libvmaf_inc, vmaf_include],
     dependencies: [stdatomic_dependency],
     c_args : [vmaf_cflags_common, compat_cflags],
-    link_with : libvmaf.get_static_lib(),
+    link_with : get_option('default_library') == 'both' ? libvmaf.get_static_lib() : libvmaf,
     install : true,
 )

